### PR TITLE
feat(ui): add types and test factories for pod.resources

### LINKED ISSUE #2345: Add types and test factories for pod.resources

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Added types and test factories for pod.resources, which follows the shape seen [here](https://pastebin.canonical.com/p/VJCMtcBdVw/)

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- N/A

## Fixes

Fixes #2345

### CODE REVIEW

File: ui/src/testing/factories/nodes.ts [hunk 1]
```diff
@@ -356,6 +365,64 @@ export const podNumaNode = define<PodNumaNode>({
...
+});
+
+export const podNumaHugepage = define<PodNumaHugepageMemory>({
+  allocated: 1024,
+  free: 2048,
+  page_size: 4068,
+});
+
+export const podNumaMemory = define<PodNumaMemory>({
+  general: {
```
Comment (huwshimi) [inline ui/src/testing/factories/nodes.ts:400]: I have a feeling it'll complain about this hard coded object.

File: ui/src/testing/factories/nodes.ts [hunk 2]
```diff
@@ -356,6 +365,64 @@ export const podNumaNode = define<PodNumaNode>({
...
+    allocated: 1024,
+    free: 2048,
+  },
+  hugepages: () => [],
+});
+
+export const podNuma = define<PodNuma>({
+  cores: {
```
Comment (huwshimi) [inline ui/src/testing/factories/nodes.ts:408]: Same thing here, it might complain that this is hardcoded.

**Review by huwshimi — APPROVED** [review]
This looks fine. A couple of tiny comments. Also, if there isn't a card for those todos it could be worth adding so you don't forget.

File: ui/src/testing/factories/nodes.ts (hunk 2 above)
Comment (Caleb-Ellis) [reply ui/src/testing/factories/nodes.ts]: Fixed here too

**Pull request merged**